### ISSUE TITLE
way to organize forked notebooks

### ISSUE THREAD

**Issue by s-u** [att/rcloud#1815]
When forking notebook from other users it may become hard to find them in your own tree. As people have more notebooks and everyone uses a different nomenclature for their paths, forked notebooks are likely to create a different structure than your own since it is simply copied. Hence it would be great if it was possible to specify a prefix for all forked notebooks such that they don't inadvertently mesh with your own notebook tree (until you explicitly want to by renaming it) - or some other option of organizing them.

Possible options:

1. have a configurable prefix (similar to new notebook prefix)
2. prepend the user name (possibly with a prefix as well) to organize forked notebooks by the person they were forked from
3. use a different color for forked notebooks
4 ... ?

I see the symptom (can't find notebooks I forked) but I think it may be worthwhile discussion possible solutions...

**Comment by gordonwoodhull** [comment]
There are some technical problems with renaming a notebook while forking it. We seem to run into GitHub concurrency issues and it doesn't know about a gist it just told us about.

If we are willing to lose the history, then we can do the prefix and fix #703/#1643 duplicated fork names, and #1715 unable to fork another user's notebook twice, at the same time.

**Comment by gordonwoodhull** [comment]
But that's not the real question, I guess. 

A more fundamental question is, does the forked notebook still belong to the original author in some way? I had always thought that once it is forked it is no different from any of your own notebooks, but apparently this is not so.

**Comment by s-u** [comment]
I see, I didn't know about the technical issue.

As for becoming your own - yes, that's true, but it still has the information that it's a fork so it is distinguishable.

As far as interaction goes, it is your own. But the problem I'm running into is that I have no  control over the name at fork time, so it gets filed somewhere in the tree that may not be anywhere close to the nomenclature I use for my own notebooks. You may say that I should simply rename it, but too often one just forks a notebook to change a line to see the result, so renaming it is the last thing on one's mind ... until you try to find it again and can't since it's not file anywhere that you would expect it.

**Comment by gordonwoodhull** [comment]
Yes, that is very messy, I agree. I also end up with some very long "limbs" in my notebook tree from forking deeply nested notebooks. I guess I would expect them to land in my root folder, but that's a special case of specifying the prefix.

Well, it kind of looks like we are moving toward ditching GitHub forks entirely, because of all the other problems, so maybe we can address all of these at once.
